### DISCUSSION TITLE
Solid-Router for non DOM (WebGL)

### DISCUSSION THREAD

**Discussion by chiefcll** [solidjs/solid-router#398] [Ideas]
Hi @ryansolid - I'm using SolidJS Universal Renderer to render to WebGL
https://github.com/lightning-js/solid

And also using this solid-router which works out of the box. I have two tweaks:

1) When you create a router can the `setupNativeEvents` be optional? Don't use anchor tags or click events at all
2) Is there a way to preload a route via useNavigate or some other function call? `router.preloadRoute(url)` - This may just need some documentation around how to do it without anchor tags.

Thanks

Chris

**Comment by ryansolid** [comment]
These are both great suggestions. I was thinking for this case we'd just setup a custom of router that didn't wire up the setupNativeEvents. It's why I separated it out that way. Basically something like this without the setupNative events: https://github.com/solidjs/solid-router/blob/main/src/routers/Router.ts

But exposing the preload routes is a good idea. I wasn't sure the best API for it but I think that it could have multiple uses. I guess technically it is already exposed in `useRouter`. The anchor tag isn't important that is just a boolean about whether it should preload data as well. You need to pass it a URL object and true and I think it should do what you want. But yeah I didn't document this as I was unsure if we wanted to provide a better API wrapper on it.

**Comment by chiefcll** [comment]
I did see how you structured it - looks straight forward to do #1.

Figuring out API's is hard work. useNavigate('url', { preload: true}) maybe? It's an important pattern to precall api's before navigating to the page and it would be easy to do on focus of the buttons. Just need a way to call it.

Thanks again!
